### DISCUSSION TITLE
iOS app crashes on startup due to missing `IActivationForViewFetcher`

### DISCUSSION THREAD

**Discussion by AArnott** [AvaloniaUI/Avalonia#16383] [Q&A] [answered]
My Avalonia app works on every desktop OS and Android. But on iOS it crashes on startup with an `ArgumentException`:

> Don't know how to detect when Nerdbank.Zcash.App.Views.MainView is activated/deactivated, you may need to implement IActivationForViewFetcher 

This happens from my `MainView` constructor, as it is chaining into the base class `ReactiveUserControl<MainViewModel>` constructor. 

I expect there's some stock implementation or solution to this for iOS. Maybe I'm just missing a package reference to something. Any suggestions?

**Comment by kekekeks** [comment, marked as answer]
You are missing `UseReactiveUI`. You can apply it to AppBuilder from your AppDelegate:
https://github.com/AvaloniaUI/Avalonia/blob/890df998a9273ec25e682585346b1e76678b4c1f/samples/ControlCatalog.iOS/AppDelegate.cs#L16-L23

(replace AfterSetup(...)` with `UseReactiveUI`)

**Reply by AArnott** [reply]
I wasn't actually missing `UseReactiveUI()`, but the trouble was (as I mention in https://github.com/AvaloniaUI/Avalonia/discussions/16382#discussioncomment-10104073) my `CustomizeAppBuilder` override was called, but had no effect on iOS.
But it's good to know once _that_ problem is solved, my `UseReactiveUI()` should solve _this_ problem.

**Reply by kekekeks** [reply]
It is called, but you are expected to reconfigure the one passed to the callback, not return a new one.

**Reply by AArnott** [reply]
Returning a new one was the only way to avoid the default constructor for my `App` class. And it works on all other platforms. So it caught me by surprise when it didn't work.
